mesonbuild/modules/pkgconfig.py

```diff
@@ -25,15 +25,65 @@
 from .. import mlog
 from ..coredata import BUILTIN_DIR_OPTIONS
 from ..dependencies import ThreadDependency
-from ..interpreterbase import permittedKwargs, FeatureNew, FeatureDeprecated, FeatureNewKwargs
-from ..interpreterbase.decorators import typed_pos_args
+from ..interpreter.type_checking import D_MODULE_VERSIONS_KW, INSTALL_DIR_KW, VARIABLES_KW
+from ..interpreterbase import FeatureNew, FeatureDeprecated
+from ..interpreterbase.decorators import ContainerTypeInfo, KwargInfo, typed_kwargs, typed_pos_args
 
 if T.TYPE_CHECKING:
+    from typing_extensions import TypedDict
+
     from . import ModuleState
     from ..interpreter import Interpreter
 
+    ANY_DEP = T.Union[
+        dependencies.Dependency, build.SharedLibrary, build.StaticLibrary,
+        build.CustomTarget, build.CustomTargetIndex, str]
+    LIBS = T.Union[
+        str, build.SharedLibrary, build.StaticLibrary, build.CustomTarget,
+        build.CustomTargetIndex]
```

Comment:
This honestly feels like something the build / dependencies subpackages should be providing, which would be useful for more than just pkgconfig.